### PR TITLE
Updates Cranelift Codegen dependence to 0.44

### PR DESCRIPTION
**Pull request by jlb6740**
In trying to build the lightbeam feature in wasmtime there are
compile errors mostly related to outdated or missing trait
definitions. This patch moves the current cranelift codegen
dependence to 0.44 in support of that, though other changes
are still needed in wasmtime to get the feature to build.

### CODE REVIEW

**Comment by jlb6740** [comment]
https://github.com/CraneStation/wasmtime/issues/389

File: src/translate_sections.rs [hunk 1]
```diff
@@ -93,7 +93,9 @@ impl binemit::RelocSink for UnimplementedRelocSink {
     ) {
         unimplemented!()
     }
-
+    fn reloc_constant(&mut self, _: binemit::CodeOffset, _: binemit::Reloc, _: ir::ConstantOffset) {
+        unimplemented!()
+    }
```
Comment (bjorn3) [inline src/translate_sections.rs:99]: Please add a newline before and after this function.

**Review by bjorn3 — APPROVED** [review]

File: src/translate_sections.rs (hunk 1 above)
Comment (jlb6740) [reply src/translate_sections.rs:99]: Thanks. Just updated.

**Comment by sunfishcode** [comment]
Lightbeam has now moved into Wasmtime, where this change is now incorporated.

**Pull request closed without merging**